walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -55,4 +53,42 @@
     }
     return validTimes;
   }
+
+  public Collection<TimeRange> findOverlapTimes(Collection<TimeRange> mandatoryTimes, Collection<TimeRange> optionalTimes, long duration) {
+    Collection<TimeRange> validTimes = new ArrayList<>();
+    for(TimeRange optional : optionalTimes) {
+      for(TimeRange mandatory : mandatoryTimes) {
+        if(mandatory.overlaps(optional)) {
+          int start = Math.max(optional.start(), mandatory.start());
+          int end = Math.min(optional.end(), mandatory.end());
+          if(end == TimeRange.END_OF_DAY) {
+            end += 1;
+          }
+          TimeRange time = TimeRange.fromStartEnd(start, end, false);
+          if(time.duration() >= duration) {
+            validTimes.add(time);
+          }
+        }
+      }
+    }
+    return validTimes;
+  }
+
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```

Comment:
Nice main method. Very easy to read without knowing how the helper methods are implemented!